js/info.js

```diff
@@ -685,12 +700,15 @@ var info = (function () {
               }
               return fields_kv;
             }
+            feature.setProperties({'fields_kv': fields_kv});
             // add a serialized version of the object so it can easily be passed through HTML GET request
             // you can deserialize it with `JSON.parse(data)` when data is the serialized data
-            feature.properties['serialized'] = function () {
-              return encodeURIComponent(JSON.stringify(feature.properties));
+            var serialized = function () {
+              return encodeURIComponent(JSON.stringify(feature.getProperties()));
             }
-            obj.features.push(feature.properties);
+            feature.setProperties({'serialized': serialized})
+            // attach ol_uid to identify feature in DOM (not all features have a feature id as property)
+            obj.features.push({...feature.getProperties(), feature_id: feature.ol_uid});
```

Comment:
> (not all features have a feature id as property)

When do you need a `feature_id` ?

> `<li id="{{feature_id}}" class="inventaire item" style="width:100%;">`

You affect `{{feature_id}}` for each `<li>` (`mst` file) but i don't really understand why. 


Because you add selected features with `mviewer.highlightFeatures(features)` and use `ol_uid` instead of `feature_id` in  `removeLayerInfo` function :

> if (feature.ol_uid == subFeature.ol_uid) {

Reply:
Yes, I need a `feature id` in the DOM to select a new feature on the map when spinning the carousel (https://github.com/geobretagne/mviewer/pull/348/files#diff-93a7855bf3f1ae2038a7d88fcd2bb8c3R413). I used the internal OL uid for this, since a feature id is not a mandatory property of a feature and can be `undefined`. I agree that using the `ol_uid` is not ideal and am open to better suggestions.

Reply:
This will be potentially broken with customlayer handle function (see cluster example).
It's fine if documented ;)

Reply:
Should work now (see below). Thanks again!